[Terra Clinical Label Value View] Link the definition list in the LabelValueView with a text provided by the consumer, that describes what the definition list is about (#875)
cerner/terra-clinical: packages/terra-clinical-label-value-view/src/terra-dev-site/doc/clinical-label-value-view/clinicalLabelValueView.1.doc.mdx

# Terra Clinical Label Value View

> The LabelValueView component should be used to create a visual and programmatic association of a label and value pair. An HTML description list is used to create the programmatic structure of the label (a term) to a value (a definition). This semantic structure is critical to support users of assistive technology. 

Author: The implementation of the definition list was done after some discussions with the UX team, who decided that this is what we want for the LabelValueView, in order to make the association of a label-value pair and for the screen readers to clearly notify the user that the 2 elements are related, even if the textValue is not always a real definition. I agree that it is not always accurate, but we didn't find any better element to use and this is what UX considered being the best way the screen readers can notify the user of what the label and the value are.
Maybe @scottwilmarth can bring more arguments than me or can explain better the need of the definition list :) Or @sdadn - do you have any suggestion of how this can be done in a better way?

> The LabelValueView component should be used to create a visual and programmatic association of a label and value pair. An HTML description list is used to create the programmatic structure of the label (a term) to a value (a definition). This semantic structure is critical to support users of assistive technology. 

Reviewer: Hi @MadalinaStelea and @sdadn  -
I think it makes sense to remove the phrases "(a term)" and "(a definition)" from the previous description I helped craft.

Also, here is some additional explanation I just wrote that speaks to reasoning behind using LabelValueView to enhance access:

"The LabelValueView component supports easy content discovery for assistive technology (AT) users, who may search for and navigate by lists on the page.  Using LabelValueView also supports consistent and clear AT delivery of the component content by supporting separate review of label/value pairs in structured announcements to AT users."

Let me know if this works better.

> The LabelValueView component should be used to create a visual and programmatic association of a label and value pair. An HTML description list is used to create the programmatic structure of the label (a term) to a value (a definition). This semantic structure is critical to support users of assistive technology. 

Author: @brooksienoodlesoup - sorry for the late response. I had some issues with the wdio in the PR, that I've just fixed. I also removed "a term" and "a definition" from the text above.

> The LabelValueView can be used as a single description list with one label value pair or a group of related label value pairs can be created.

Reviewer 2: Was this change committed?

- To use as a description list with a single label value pair, ensure the `isChildOfDescriptionList` prop to `false`
- If you have a group of related label value pairs all related to a single topic, creating one description list `<DL>` to enclose all of the related label value pairs is the best practice. The LabelValueView can be included in an already existing description list by setting the `isChildOfDescriptionList` prop to `true` in which case the parent `<DL>` tag must be provided in another component that consumes LabelValueView or added separately.

It is highly recommended that the consumer add text on the page, often set as a heading, to explain the topic of the description.

> LabelValueViews and the description lists that enclose them are often visually associated with a topic on the page. It is best practice to programmatically associate the LabelValueView with the ID of the topic on the page using one of the following methods:

Reviewer 2: ```suggestion
LabelValueViews and the encapsulating description lists are often visually associated with a topic on the page. It is best practice to programmatically associate the LabelValueView with the ID of the element that labels it on the page by using one of the following methods:
```

- If the `isChildOfDescriptionList` prop is set to `false`, associate the list by setting the `topicTextId` prop to the ID of the topic.

> - If the `isChildOfDescriptionList` prop is set to `true` please associate the description list with the topic text using the `aria-labelledby` attribute on the description list that points to the ID of the topic.

Reviewer 2: ```suggestion
- If the `isChildOfDescriptionList` prop is set to `true` associate the description list with the label using the `aria-labelledby` attribute on the description list that points to the ID of the topic.
```

# Terra Clinical Label Value View

The LabelValueView component creates a visual and programmatic association of a label-value pair. An HTML description list is used to create the programmatic structure of the label to a value. This semantic structure is critical to support users of assistive technology. 

> LabelValueView can be used as a single description list with one label-value pair or as a group of related label-value pairs.

Reviewer 3: Can we add two short code examples in this doc to illustrate the different usages?

Author: I have already updated the examples in the previous Jira, to ilustrate the 2 different usage options of the LabelValueView - as a definition list or as individual term-definition pair. In the current PR, I have also set the aria-labelledby/ariaLabelledBy props for the existing examples.

The LabelValueView can be used as a single description list with one label value pair or a group of related label value pairs can be created.
- To use as a description list with a single label value pair, ensure the `isChildOfDescriptionList` prop to `false`
- If you have a group of related label value pairs all related to a single topic, creating one description list `<DL>` to enclose all of the related label value pairs is the best practice. The LabelValueView can be included in an already existing description list by setting the `isChildOfDescriptionList` prop to `true` in which case the parent `<DL>` tag must be provided in another component that consumes LabelValueView or added separately.

It is highly recommended that the consumer add text on the page, often set as a heading, to explain the topic of the description.

LabelValueViews and the description lists that enclose them are often visually associated with a topic on the page. It is best practice to programmatically associate the LabelValueView with the ID of the topic on the page using one of the following methods:

> - If the `isChildOfDescriptionList` prop is set to `false`, associate the list by setting the `topicTextId` prop to the ID of the topic.

Author: Sure, I've made the change. Thank you for bringing it up, I missed a few places where I should have updated it.
